lib/galaxy/tool_util_models/parameters.py

```diff
@@ -338,17 +341,30 @@ def request_requires_value(self) -> bool:
 TestCaseDataSrcT = Literal["File"]
 
 
-class DataRequestHda(StrictModel):
+class LegacyRequestModelAttributes(StrictModel):
+    # Here for bioblend's sake, should be stripped
```

Comment:
Is there any chance we can clean this before we create the models? I don't see API changes but maybe they are coming through implicitly? I hate this but also if the reality is it is needed here - I can live with it.

Reply:
You mean popping these out of the raw dictionary ?

> I don't see API changes but maybe they are coming through implicitly?

bioblend just sets these on reruns, which I think is a bioblend bug, but I guess we also don't want to break the clients ?

> I hate this but also if the reality is it is needed here - I can live with it.

My thinking is that this has relatively little impact, and all properly typed clients (I suppose that's only our own typescript client at this point) would refuse to send these fields (since they aren't part of the generated schema)